src/core/Lockfile_target.ml

```diff
@@ -0,0 +1,14 @@
+type t = {
```

Comment:
Can you add a small comment explaining what is the difference between a lockfile target and a manifest target?
Maybe with some concrete examples?